src/ui/dialogs/omemoDevices.ts

```diff
@@ -39,15 +40,26 @@ class OmemoDeviceDialog {
 
          dom.find('.jsxc-omemo-peerdevices, .jsxc-omemo-owndevices').empty();
 
-         this.insertDevices(peerDevices, identityManager, dom.find('.jsxc-omemo-peerdevices'));
-         this.insertDevices(ownDevices, identityManager, dom.find('.jsxc-omemo-owndevices'));
-
+         this.insertDevices(contact.getJid().bare, peerDevices, identityManager, dom.find('.jsxc-omemo-peerdevices'));
+         this.insertDevices(contact.getAccount().getJID().bare, ownDevices, identityManager, dom.find('.jsxc-omemo-owndevices'));
          if (ownDevices.length > 1) {
             this.addCleanUpAction();
          }
+
       });
    }
 
+   private buildQRCode(jid: string, deviceid: string, key: string)
```

Comment:
The device name is the bare jid, so we can just pass the device object.

Reply:
changed